### ISSUE TITLE
Environment variables limited to 256 characters

### ISSUE THREAD

**Issue by jmahlik** [aws/sagemaker-python-sdk#2932]
**Describe the bug**
Currently, the value length constraints for environment variables are quite limited in `CreateProcessingJob` and `CreateTrainingJob`. 

* [CreateProcessingJob](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_CreateProcessingJob.html#sagemaker-CreateProcessingJob-request-Environment): 256
* [CreateTrainingJob](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_CreateTrainingJob.html#sagemaker-CreateTrainingJob-request-Environment): 512 

This causes issues when passing urls or other semi-sensitive variables via the environment parameter, as they often exceed the limit. I am not aware of an alternative to pass items that are sensitive in nature to a job at runtime. Storing semi-sensitive info on  s3 and treating it as an input cannot be used as a workaround, as variables should not be shared with everyone who has access to the s3 bucket. ex. api tokens. 

It looks like it would be possible to have an api call accept keys that are over 256 characters in length. Example https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_CreateTrainingJob.html#sagemaker-CreateTrainingJob-request-HyperParameters accepts 2500 characters.

**To reproduce**
Pass a string over the char limit similar to below.

```python
estimator = sagemaker.estimator.Estimator(
    image,
    role,
    environment={"LONG_STRING": "".join(("0" for _ in range(513)))},
    instance_count=1,
    instance_type=instance_type,
)

script_processor = sagemaker.processing.ScriptProcessor(
    command=["my", "command"],
    env={"LONG_STRING": "".join(("0" for _ in range(257)))},
    image_uri=image,
    role=role,
    instance_count=1,
    instance_type=instance_type,
)

```

**Expected behavior**
Ability to pass longer environment variables to jobs.

**Screenshots or logs**

```console
botocore.exceptions.ClientError: An error occurred (ValidationException) when calling the CreateProcessingJob operation: 1 validation error detected: Value '<CENSORED>' at 'environment' failed to satisfy constraint: Map value must satisfy constraint: [Member must have length less than or equal to 256, Member must have length greater than or equal to 0, Member must satisfy regular expression pattern: [\S\s]*]
```

**System information**
A description of your system. Please provide:
- **SageMaker Python SDK version**:  2.52.0
- **Framework name (eg. PyTorch) or algorithm (eg. KMeans)**: N/A
- **Framework version**: N/A
- **Python version**: 3.9
- **CPU or GPU**: CPU
- **Custom Docker image (Y/N)**: Both

**Additional context**
I suspect others may be facing the same issue as there is mention of passing custom urls for pip configurations. 

https://github.com/aws/sagemaker-python-sdk/issues/2207#issuecomment-810096332

**Comment by fxmarty** [comment]
It is not clear why there is such a limit.

**Comment by mesquita** [comment]
Was anyone able to get past this problem?

**Comment by jmahlik** [comment]
> Was anyone able to get past this problem?

No, it's enforced by the api. Writing files to s3 or using secrets manager are about the only solutions.
